docs/modules/serialization/pages/compact-serialization.adoc

```diff
@@ -35,368 +35,1620 @@ a class to implement a particular interface. Serializers might be implemented an
+            e.age = 0;
+        }
+        
+        return e;
+    }
+};
+----
+--
+
+Node.js::
++
+--
+[source,javascript]
+----
+class EmployeeSerializer extends CompactSerializer {
+    read(reader){
+        const id = reader.readInt64('id');
+        const name = reader.readString('name');
+        // Read the "age" if it exists, or use the default value 0.
+        // reader.readInt32("age") would throw if the "age" field
+        // does not exist in data.
+        let age;
+        if (reader.getFieldKind("age") == FieldKind.INT32) {
+            age = reader.readInt32("age");
+        } else {
+            age = 0;
+        }
+        return new Employee(id, name, age);
+    }
+}
+----
+--
+
+Python::
++
+--
+[source,python]
+----
+class EmployeeSerializer(CompactSerializer[Employee]):
+    def read(self, reader: CompactReader):
+        id = reader.read_int64("id")
+        name = reader.read_string("name")
+        # Read the "age" if it exists, or use the default value 0.
+        # reader.read_int32("age") would throw if the "age" field
+        # does not exist in data.
+        if reader.get_field_kind("age") == FieldKind.INT32:
+            age = reader.read_int32("age")
+        else:
+            age = 0
+        return Employee(id, name, age)
+    ...
+----
+--
+
+Go::
++
+--
+[source,go]
+----
+func (em EmployeeSerializer) Read(reader serialization.CompactReader) any {
+    id := reader.ReadInt64("id")
+    name := reader.ReadString("name")
+    // Read the "age" if it exists, or use the default value 0.
+    // reader.ReadInt32("age") would panic if the "age" field
+    // does not exist in data.
+    var age int32
+    if reader.getFieldKind("age") == serialization.FieldKindInt32 {
+        age = reader.ReadInt32("age")
+    }
+    return &Employee{ID: id, Name: name, Age: age}
+}
+
+----
+--
+
+====
+
+Note that, when an old reader reads data written by an old writer, or a new reader reads a data
+written by a new writer, they will be able to read all fields written.
+
+One thing to be careful while evolving the class is to not have any conditional code
+in the `write` method. That method must write all the fields available in the current version
+of the class to the writer, with appropriate field names and types. Hazelcast uses the `write`
+method of the serializer to extract a schema out of the object, hence any conditional code
+that may or may not run depending on the object in that method might result in an undefined
+behavior.
 
 == GenericRecord Representation
 
-Compact serialized objects
-can also be represented by a `GenericRecord`, without requiring the class in the classpath. See xref:clusters:accessing-domain-objects.adoc[].
+Compact serialized objects can also be represented by a `GenericRecord`, without requiring
+the class in the classpath. See xref:clusters:accessing-domain-objects.adoc[].
+
+== SQL Support
+
+Compact serialized objects can be used in SQL statements, provided that a mappings are created,
```

Comment:
```suggestion
Compact serialized objects can be used in SQL statements, provided that a mapping is created,
```

Reply:
Changed it to `provided that mappings are created`